attendance-portal/src/Components/Details.js

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { useEffect, useState } from 'react';
+import { getStudentDetails } from '../redux/api';
+import ReactJson from 'react-json-viewer'
+
+
+const Details = () => {
+    const [initialState, setInitialState] = useState([])
+    useEffect(() => {
+        getStudentDetails().then(res => {
+            console.log(res);
+            return (
+                res.data
+            )
+        }).then(jsonResponse => setInitialState(jsonResponse))
+    }, [])
```

Comment:
Use `async` and `await` and asynchronous programming instead of `then()` for promises